src/controllerclientimpl.cpp

```diff
@@ -1675,7 +1675,7 @@ void ControllerClientImpl::_UploadFileToController_UTF8(const std::string& filen
     }
 
     MUJIN_LOG_DEBUG(str(boost::format("upload %s")%uri))
-    _UploadFileToControllerViaForm(fin, filenameoncontroller, _baseuri + "fileupload");
+    _UploadFileToControllerViaForm(fin, filenameoncontroller, _baseuri + "fileupload?filename=" + filenameoncontroller);
```

Comment:
Don't you need to encode this query string?

Reply:
(let me check http)

Reply:
@ziyan _PrepareDestinationURI_UTF8 calls `_EncodeWithoutSeparator(s)`, so uri (and hence filenameoncontroller) are encoded. I confirmed by wireshark. #79's intention was to disable this _EncodeWithoutSeparator conditionally.

Reply:
_PrepareDestinationURI_UTF8's role is currently unclear.

Now that webdav is gone, _PrepareDestinationURI_UTF8/16's second argument should be always false. I reimplemented to UploadDataToController_UTF8/16 methods to reuse _UploadFileToControllerViaForm. It is available on recoverUploadData branch. Let's work on it after merging this #107.